google-cloud-firestore/lib/google/cloud/firestore/document_snapshot.rb

```diff
@@ -0,0 +1,367 @@
+# limitations under the License.
+
+
+require "google/cloud/firestore/v1beta1"
+require "google/cloud/firestore/document_reference"
+require "google/cloud/firestore/collection_reference"
+require "google/cloud/firestore/convert"
+
+module Google
+  module Cloud
+    module Firestore
+      ##
+      # # DocumentSnapshot
+      #
+      # A document snapshot object is an immutable representation for a
+      # document in a Cloud Firestore database.
+      #
+      # The snapshot can reference a non-existing document.
+      #
+      # @example
+      #   require "google/cloud/firestore"
+      #
+      #   firestore = Google::Cloud::Firestore.new
+      #
+      #   # Get a document with data
+      #   nyc_snap = firestore.doc("cities/NYC").get
+      #
+      #   # Get the document data
+      #   nyc_snap[:population] #=> 1000000
+      #
+      class DocumentSnapshot
+        ##
+        # @private The Google::Firestore::V1beta1::Document object.
+        attr_accessor :grpc
+
+        ##
+        # The document identifier for the document with data.
+        #
+        # @return [String] document identifier.
+        def document_id
+          ref.document_id
+        end
+
+        ##
+        # A string representing the path of the document, relative to the
+        # document root of the database.
+        #
+        # @return [String] document path.
+        def document_path
+          ref.document_path
+        end
+
+        ##
+        # @private A string representing the full path of the document resource.
+        #
+        # @return [String] document resource path.
+        def path
+          ref.path
+        end
+
+        # @!group Access
+
+        ##
+        # The document reference object for the data.
+        #
+        # @return [DocumentReference] document reference.
+        #
+        # @example
+        #   require "google/cloud/firestore"
+        #
+        #   firestore = Google::Cloud::Firestore.new
+        #
+        #   # Get a document with data
+        #   nyc_snap = firestore.doc("cities/NYC").get
+        #
+        #   # Get the document reference
+        #   nyc_ref = nyc_snap.ref
+        #
+        def ref
+          @ref
+        end
+        alias_method :reference, :ref
+
+        ##
+        # Retrieves a list of collections nested under the document with data.
+        #
+        # @yield [collections] The block for accessing the collections.
+        # @yieldparam [CollectionReference] collection A collection.
+        #
+        # @return [Enumerator<CollectionReference>] collection list.
+        #
+        # @example
+        #   require "google/cloud/firestore"
+        #
+        #   firestore = Google::Cloud::Firestore.new
+        #
+        #   nyc_snap = firestore.doc("cities/NYC").get
+        #
+        #   nyc_snap.cols.each do |col|
+        #     puts col.collection_id
+        #   end
+        #
+        def cols &block
```

Comment:
DocumentSnapshots should only provide access to the data in a document as well as its DocumentReference (and maybe path).

Reply:
Removed.